rest/oidc_api_test.go

```diff
@@ -1,139 +1,612 @@
+			name:             "oidc default provider specified but current provider is not default",
+			inputRequestURL:  "/db/_oidc?provider=facebook&offline=true",
+			wantProviderName: "facebook",
+		}, {
+			name:             "oidc challenge default provider specified but current provider is not default",
+			inputRequestURL:  "/db/_oidc_challenge?provider=facebook&offline=true",
+			wantProviderName: "facebook",
+			expectedProvider: "provider%3Dfacebook",
+			oidcChallenge:    true,
+		}, {
+			// When multiple providers are defined, default provider is specified and the current provider is
+			// default, then current provider should NOT be added to the generated OpenID Connect callback URL.
+			name:            "oidc default provider specified and current provider is default",
+			inputRequestURL: "/db/_oidc?provider=google&offline=true",
+		}, {
+			name:            "oidc challenge default provider specified and current provider is default",
+			inputRequestURL: "/db/_oidc_challenge?provider=google&offline=true",
+			oidcChallenge:   true,
+		}, {
+			// When multiple providers are defined, default provider is specified and no current provider is
+			// provided, then provider name should NOT be added to the generated OpenID Connect callback URL.
+			name:            "oidc default provider specified with no current provider",
+			inputRequestURL: "/db/_oidc?offline=true",
+		}, {
+			name:            "oidc challenge default provider specified with no current provider",
+			inputRequestURL: "/db/_oidc_challenge?offline=true",
+			oidcChallenge:   true,
+		},
+	}
 
-	t.Run("default provider configured and current provider is default", func(t *testing.T) {
-		// When multiple providers are defined, default provider is specified and the current provider is
-		// default, then current provider should NOT be added to the generated OpenID Connect callback URL.
-		resp := rt.SendAdminRequest(http.MethodGet, "/db/_oidc?provider=google&offline=true", "")
-		require.Equal(t, http.StatusFound, resp.Code)
-		location := resp.Header().Get(headerLocation)
-		require.NotEmpty(t, location, "Location should be available in response header")
-		locationURL, err := url.Parse(location)
-		require.NoError(t, err, "Location header should be a valid URL")
-		redirectURI := locationURL.Query().Get(authRedirectURI)
-		require.NotEmpty(t, location, "redirect_uri should be available in auth URL")
-		redirectURL, err := url.Parse(redirectURI)
-		require.NoError(t, err, "redirect_uri should be a valid URL")
-		assert.Equal(t, "", redirectURL.Query().Get(authProvider))
-	})
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.oidcChallenge {
+				resp := rt.SendAdminRequest(http.MethodGet, tc.inputRequestURL, "")
+				require.Equal(t, http.StatusUnauthorized, resp.Code)
+				wwwAuthHeader := resp.Header().Get("Www-Authenticate")
+				assert.Contains(t, wwwAuthHeader, tc.expectedProvider)
```

Comment:
Shouldn't we be verifying the redirectURI in the challenge case, as we're doing below for the oidc case?